### PR TITLE
chore: code smell fixes and lint rule enablement

### PR DESCRIPTION
**Pull request by Klaus-Keller**
# Issue
Sonar reported code smells.

## Description
Fix sonar reported code smells. To avoid running into the same issues again, the following two linting rules have been activated:

```
@typescript-eslint/no-redundant-type-constituents
@typescript-eslint/prefer-regexp-exec
```

## Checklist for Pull Requests

- [x] Supplied as many details as possible on this change
- [x] Included the link to the associated issue in the Issue section above
- [x] The code conforms to the general [development guidelines](https://github.com/SAP/guided-answers-extension/blob/main/docs/developer-guide.md).
- [x] The code has **unit tests** where applicable and is easily unit-testable
- [x] This branch is appropriately named for the associated issue

### CODE REVIEW

File: packages/types/src/types.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ export interface GuidedAnswerAPI {
 export interface GuidedAnswersFeedback {
     treeId: GuidedAnswerTreeId;
     nodeId: GuidedAnswerNodeId;
-    message: 'Solved' | 'Not Solved' | string;
+    message: string; // if used to sent feedback outcome message is set to  'Solved' | 'Not Solved'
```
Comment (theStiggySpd) [inline packages/types/src/types.ts:113]: Please change the comment. Doesn't sound right.

**Review by theStiggySpd — CHANGES_REQUESTED** [review]
Very minor comment @Klaus-Keller

File: packages/types/src/types.ts (hunk 1 above)
Comment (Klaus-Keller) [reply packages/types/src/types.ts]: @theStiggySpd, you are right it sounds confusing. I removed the comment completely. The fact that it needs to be set to `Solved` or `Nor Solved` is implemented  behind the API function and the API expects `solved: boolean`, so it is clear to consumers and does not need additional comments:
https://github.com/SAP/guided-answers-extension/blob/6419ba932f868a66ad4d5c7110db85e696bef81a/packages/core/src/guided-answers-api.ts#L302-L310

Comment (theStiggySpd) [reply packages/types/src/types.ts]: yes i saw that. Thanks

**Review by theStiggySpd — APPROVED** [review]
- changeset looks good
- Changes in package.json looks correct
- Code changes based on the new linting rules looks correct too
*No local testing done

**Pull request merged**